### PR TITLE
Update cirq to bloq interop to automatically replace basic gates with their Bloq versions

### PR DESCRIPTION
**Pull request by tanujkhattar**
Pulled out from https://github.com/quantumlib/Qualtran/pull/732

TODO(as part of a follow-up PR): @mpharrigan Can we add a method to convert Cirq's `AbstractControlValues` to `CtrlSpec`? This would allow converting `ControlledGate` instaces to `Controlled` Bloq. There's also a nuance that both `cirq.Gate.controlled()` and `Bloq.controlled()`exist and they have different interfaces so cirq style API and bloq style API clashes for `GateWithRegisters`.

### CODE REVIEW

File: qualtran/cirq_interop/_cirq_to_bloq.py [hunk 1]
```diff
@@ -300,29 +300,80 @@ def _gather_input_soqs(
...
+    if gate == cirq.CNOT:
+        return CNOT()
+    if gate == cirq.TOFFOLI:
+        return Toffoli()
+    if gate == cirq.X:
+        return XGate()
+    if gate == cirq.Z:
+        return ZGate()
+    if gate == cirq.SWAP:
+        return TwoBitSwap()
```
Comment (mpharrigan) [inline qualtran/cirq_interop/_cirq_to_bloq.py:346]: could be a dictionary lookup

File: qualtran/cirq_interop/_cirq_to_bloq.py [hunk 2]
```diff
@@ -300,29 +300,80 @@ def _gather_input_soqs(
...
+    if isinstance(gate, cirq.Rx):
+        return Rx(angle=gate._rads)
+    if isinstance(gate, cirq.Ry):
+        return Ry(angle=gate._rads)
+    if isinstance(gate, cirq.XPowGate):
+        return XPowGate(exponent=gate.exponent, global_shift=gate.global_shift)
+    if isinstance(gate, cirq.YPowGate):
+        return YPowGate(exponent=gate.exponent, global_shift=gate.global_shift)
+    if isinstance(gate, cirq.ZPowGate):
+        return ZPowGate(exponent=gate.exponent, global_shift=gate.global_shift)
```
Comment (mpharrigan) [inline qualtran/cirq_interop/_cirq_to_bloq.py:360]: these (along with CZPowGate) can probably (?) be condensed

```python
if isinstance(gate, (Rx, Ry, Rz):
  return BLOQMAP[gate.__class__](angle=gate._rads)
if isintance(gate, (XPowGate, YPowGate, ...)):
  return BLOQMAP[gate.__class__](exponent=gate.exponent, global_shirt=gate.global_shift)
```

**Review by mpharrigan — APPROVED** [review]
Left some ideas for performance tweaks, but I don't know if they would be faster in practice

**Comment by tanujkhattar** [comment]
Instantiating a local dictionary on every function call would not be faster, but I think it's cleaner so I have done the replacement. Merging now!

**Comment by mpharrigan** [comment]
can't you pull out the dictionary as a (private) global?

**Comment by tanujkhattar** [comment]
I'll have to import all the basic gates, that can probably lead to circular imports, so I wanted to keep the global imports minimal. I'm sure I can find a way around if it becomes a bottleneck.

**Comment by mpharrigan** [comment]
ah fair point

**Pull request merged**